src/lib/logical_query_plan/join_node.cpp

```diff
@@ -57,17 +57,16 @@ std::vector<std::shared_ptr<AbstractExpression>> JoinNode::output_expressions()
    */
 
   const auto& left_expressions = left_input()->output_expressions();
-  const auto& right_expressions = right_input()->output_expressions();
-
   const auto output_both_inputs =
       join_mode != JoinMode::Semi && join_mode != JoinMode::AntiNullAsTrue && join_mode != JoinMode::AntiNullAsFalse;
+  if (!output_both_inputs) return left_expressions;
```

Comment:
I like this guideline but to be fair:

```
Searching 643 files (src/lib) for "if \([\w]*\) return" (regex)
...
16 matches across 11 files
```

Reply:
I think there are other guidelines that are not rigorously followed. We might want to have a discussion on how to deal with that in the Fortnightly.

Reply:
Update here: the regex `if \([\w]*\) return` is not sufficient. First, `[\w]*` does not account for comparisons/operators , parantheses and dots of of argument/method calls, etc. Second, if statements can be followed by any other keyword than `return`, such as `continue`, `break`, or even operations and method calls.
I fiddled around with `if \([^{^\n]+\)\s[^{^\n^&^\|]+\n`. Basically thats a line with an if that has anything but but opening curly bracket or newline in the condition,  followed by something that is neither `{`, newline, or `&` and `|` (the latter for multi line logical expressions with logical and/or) and a final newline. Also, I searched `src` rather than `lib`. Numbers on that are slightly different.
```
Searching 981 files for "if \([^{^\n]+\)\s[^{^\n^&^\|]+\n" (regex)
...
552 matches across 192 files
```
Although I'll add the braces for if statements introduced in this PR, applying the style everywhere is either something for an own PR or something we pretend we haven't seen.